prowler/providers/aws/services/kms/kms_service.py

```diff
@@ -43,73 +48,93 @@ def _list_keys(self, regional_client):
+                f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
+            )
+
+    def _get_key_rotation_status(self):
+        logger.info("KMS - Get Key Rotation Status...")
+        try:
+            for key in self.keys:
+                if (
+                    key.origin
+                    and key.manager
+                    and "EXTERNAL" not in key.origin
+                    and "AWS" not in key.manager
+                ):
+                    regional_client = self.regional_clients[key.region]
+                    try:
+                        key.rotation_enabled = regional_client.get_key_rotation_status(
+                            KeyId=key.id
+                        )["KeyRotationEnabled"]
+                    except Exception as error:
+                        logger.error(
+                            f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
+                        )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
+            )
 
     def _get_key_policy(self):
         logger.info("KMS - Get Key Policy...")
-        for key in self.keys:
-            if (
-                key.manager and key.manager == "CUSTOMER"
-            ):  # only customer KMS have policies
-                regional_client = self.regional_clients[key.region]
-                try:
-                    key.policy = json.loads(
-                        regional_client.get_key_policy(
-                            KeyId=key.id, PolicyName="default"
-                        )["Policy"]
-                    )
-                except Exception as error:
-                    logger.error(
-                        f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
-                    )
+        try:
+            for key in self.keys:
+                if (
+                    key.manager and key.manager == "CUSTOMER"
+                ):  # only customer KMS have policies
+                    regional_client = self.regional_clients[key.region]
+                    try:
+                        key.policy = json.loads(
+                            regional_client.get_key_policy(
+                                KeyId=key.id, PolicyName="default"
+                            )["Policy"]
+                        )
+                    except Exception as error:
+                        logger.error(
+                            f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
+                        )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
+            )
```

Comment:
@MrCloudSec Same here. The outer `try-except` is redundant, right?

Reply:
No, we are covering the rest of the lines.